lambda_src/lambda_function.py

```diff
@@ -155,30 +207,59 @@ def sync_flow(event: Any, context: Any = None) -> Dict[Text, Any]:
             'statusCode': 200,
             'body': b64encode(compress(data_dumps.encode())).decode(),
             'isBase64Encoded': True,
-            'headers': {'Content-Encoding': 'gzip'}
+            'headers': {'Content-Encoding': 'gzip'},
         }
+        end_time = timer()
+        if (
+            response
+            and (end_time - start_time) > RESPONSE_TIME_BEFORE_LOCK_CACHE_STORAGE
+        ):
+            LOG.debug('Storing the response in lock cache.')
+            finish_batch_processing(batch_id, response, res_data)  # write the response
+
+    response_length = len(dumps(response))
+    if response_length > RESPONSE_SIZE_LIMIT:
+        response = construct_size_error_response(response_length, req_body)
+    return response
 
-    if len(response) > 6_000_000:
-        response = dumps(
-            {
-                'data': [
-                    [
-                        rn,
-                        {
-                            'error': (
-                                f'Response size ({len(response)} bytes) will likely'
-                                'exceeded maximum allowed payload size (6291556 bytes).'
-                            )
-                        },
-                    ]
-                    for rn, *args in req_body['data']
+
+def construct_size_error_response(
+    size_exceeded_response_length: int, req_body: Dict[Text, Any]
+) -> ResponseType:
+    """
+    Creates a new response object with an error message,
+    for when the response size is likely to exceed the allowed payload size.
+
+    Args:
+        size_exceeded_response (int): Response object to calculate the size.
+        req_body (Dict[Text, Any]): Body of the request, obtained from the events object.
+
+    Returns:
+        ResponseType: Represents the response with the error message.
+    """
+    error_dumps = dumps(
+        {
+            'data': [
+                [
+                    rn,
+                    {
+                        'error': (
+                            f'Response size ({size_exceeded_response_length} bytes) '
+                            'exceeded maximum allowed payload size (6291556 bytes).'
```

Comment:
either remove the global variable above bytes count above or use it here

Reply:
Not sure if I understood this one actually, could you elaborate a bit more please?